assets/src/amp-story-editor-blocks.js

```diff
@@ -179,35 +178,37 @@ const wrapperWithSelect = compose(
  * @param {Object} BlockListBlock BlockListBlock element.
  * @return {Function} Handler.
  */
-const addWrapperProps = createHigherOrderComponent(
-	( BlockListBlock ) => {
-		return wrapperWithSelect( ( props ) => {
-			const { blockName, hasSelectedInnerBlock, attributes } = props;
-
-			// If we have an inner block selected let's add 'data-amp-selected=parent' to the wrapper.
-			if (
-				hasSelectedInnerBlock &&
-				(
-					'amp/amp-story-page' === blockName
-				)
-			) {
-				return <BlockListBlock { ...props } data-amp-selected={ 'parent' } />;
-			}
-
-			// If we got this far and it's not an allowed inner block then lets return original.
-			if ( -1 === ALLOWED_BLOCKS.indexOf( blockName ) ) {
-				return <BlockListBlock { ...props } />;
-			}
-
-			return <BlockListBlock
-				{ ...props }
-				data-amp-image-caption={ ( 'core/image' === blockName && ! attributes.ampShowImageCaption ) ? 'noCaption' : undefined }
-				data-font-family={ attributes.ampFontFamily || undefined }
-			/>;
-		} );
-	},
-	'addWrapperProps'
-);
+const addWrapperProps = ( BlockListBlock ) => {
+	return wrapperWithSelect( ( props ) => {
+		const { blockName, hasSelectedInnerBlock, attributes } = props;
+
+		// If it's not an allowed block then lets return original;
+		if ( -1 === ALLOWED_BLOCKS.indexOf( blockName ) ) {
+			return <BlockListBlock { ...props } />;
+		}
+
+		let wrapperProps;
+
+		// If we have an inner block selected let's add 'data-amp-selected=parent' to the wrapper.
+		if (
+			hasSelectedInnerBlock &&
+			(
+				'amp/amp-story-page' === blockName
+			)
+		) {
+			wrapperProps = Object.assign( {}, props.wrapperProps, { 'data-amp-selected': 'parent' } );
```

Comment:
Why using `Object.assign` instead of just using `<BlockListBlock { ...props } data-amp-selected={ 'parent' } />`? I'd want to be consistent with using JSX and object spread here.

Reply:
The `data-amp-selected={ 'parent' }` needs to be part of `props.wrapperProps`, not `props`.
Here the `wrapperProps` is being created of the previously existing + the new properties and then later used within `<BlockListBlock { ...props } wrapperProps={ wrapperProps } />`.

Thoughts?

Reply:
Gotcha, makes sense. Still could use spread operator though for that :-)

Reply:
Good point!